woocommerce/woocommerce-gateway-stripe
# Move the UPE action hooks to the IntentsController

# Changes proposed in this Pull Request:

Issue: #1855

This PR groups all UPE action hooks from `WC_Stripe_Intent_Controller` and `WC_Stripe_Order_Handler` in `WC_Stripe_Intent_Controller` and avoid creating new instances of the UPE Gateway, using `get_gateway()` and checking the class of the active `stripe` gateway.

The reason for this fix is to only include the UPE javascript when UPE is enabled.

# Testing instructions

- Make sure UPE is disabled (`Enable new checkout experience` is unchecked in Settings)
- Add any product to the cart.
- Go to the checkout, and check that the file `upe_classic.js` is not loaded.
- Make sure UPE is `enabled` in Settings
- Add any product to the cart.
- Go to the checkout, and check that this time the file `upe_classic.js` is loaded.
- Complete the checkout using any test Credit Card.
